tests/functional/conftest.py

```diff
@@ -35,7 +26,6 @@ def pyramid_app():
 
 @pytest.fixture
 def app(pyramid_app, db_engine):
-    _clean_database(db_engine)
     db.init(db_engine)
```

Comment:
Does this need to be like this?

```suggestion
    db.init(db_engine, drop=True)
```

As the `db_engine()` fixture is scoped to the session, but `app()` isn't and needs to be cleared between runs (as we potentially setup a bunch of stuff for each test).

Reply:
That would work but only for tests that actually use the `app` fixture, which admittedly is probably most or all functests. We could make the `app` fixture `autouse=True` or move the `db.init(db_engine, drop=True)` into a separate `db_init` fixture with `autouse=True`. But I don't think we want to do that because recreating the whole DB schema is slow. We do want to recreate it at the beginning of each test run: the DB schema defined by the models might have changed since the last test run, so we want to recreate the schema from scratch based on the current models, before each test run. But between each test we just want to undo any data changes that might have been made by the previous test, not make any schema changes, because that's much faster. That's why h uses a `clean_database()` function that uses `TRUNCATE`: it's the fastest way to clean the DB between tests